lib/fs/util.go

```diff
@@ -89,3 +89,49 @@ func IsParent(path, parent string) bool {
 	}
 	return strings.HasPrefix(path, parent)
 }
+
+func CommonPrefix(first, second string) string {
+	if filepath.IsAbs(first) != filepath.IsAbs(second) {
+		// Whatever
+		return ""
+	}
+
+	firstParts := strings.Split(filepath.Clean(first), string(PathSeparator))
+	secondParts := strings.Split(filepath.Clean(second), string(PathSeparator))
+
+	isAbs := filepath.IsAbs(first) && filepath.IsAbs(second)
```

Comment:
Just so I have something to remark: After the first check above you wouldn't have to check both :P

Reply:
Why not? I want to know if both are non absolute. On unix it checks if first rune is / so this is pretty cheapo.

Reply:
You already check whether the absolutemess of both paths is the same before, so checking just one path gives the same result

Reply:
Single instruction versus "why on earth"

Reply:
Yep, and I shouldn't have wrote it in the first place, this discussion I started just wastes time over nothing.